includes/class-woocommerce.php

```diff
@@ -274,165 +281,73 @@ private function is_request( $type ) {
 			case 'admin':
 				return is_admin();
 			case 'ajax':
-				return defined( 'DOING_AJAX' );
+				return wp_doing_ajax();
 			case 'cron':
 				return defined( 'DOING_CRON' );
```

Comment:
You also can use https://developer.wordpress.org/reference/functions/wp_doing_cron/ since 4.8